libsolidity/formal/Predicate.cpp

```diff
@@ -243,9 +243,16 @@ string Predicate::formatSummaryCall(vector<smtutil::Expression> const& _args) co
 		fun->isFallback() ? "fallback" :
 		fun->isReceive() ? "receive" :
 		fun->name();
-	solAssert(fun->annotation().contract, "");
-	return fun->annotation().contract->name() + "." + fName + "(" + boost::algorithm::join(functionArgs, ", ") + ")" + value;
 
+	string prefix;
+	if (fun->isFree())
+		prefix = fun->sourceUnitName() + "::";
```

Comment:
I am fine with this notation. 
Maybe having something like `file<a.sol>::f(1, 2)`, or `file("a.sol")::f(1,2)`is also fine.
If it looks better to you we can change, if not we should leave as is.

Reply:
I'd prefer to have just the filename. Thinking again, maybe `file:fun` (`:` instead of `::`) makes more sense since that's the convention for other things